### PR TITLE
MAYA-105537 Add CPV support and curveBasis support for basiscurves complexity level 1

### PR DESCRIPTION
**Pull request by HdC-adsk**
* Supported CPV for basisCurves complexity level 1
* Supported curve basis functions (including Bezier, BSpline and CatmullRom) for cubic basisCurves complexity level 1.
* Disabled complexity level 1 drawing for basisCurves on Maya 2018/2019/2020 because the necessary Maya API to allow plug-ins to build a complete tessellation shader graphs are not included in those releases.

### CODE REVIEW

**Comment by HdC-adsk** [comment]
The build errors in preflight are all related to the recent "SharedPtrBase" API addition which should be fixed separately.

File: lib/mayaUsd/render/vp2RenderDelegate/render_delegate.cpp [hunk 1]
```diff
@@ -289,12 +349,14 @@ namespace
...
         //! Shader registry used by fallback shaders
-        MShaderMap              _fallbackShaders[static_cast<size_t>(FallbackShaderType::kCount)];
+        MShaderMap              _fallbackShaders[FallbackShaderTypeCount];
         MShaderMap              _3dSolidShaders;
 
-        MHWRender::MShaderInstance*  _fallbackCPVShader { nullptr }; //!< Fallback shader with CPV support
-        MHWRender::MShaderInstance*  _3dFatPointShader { nullptr };  //!< 3d shader for points
-        MHWRender::MShaderInstance*  _3dCPVSolidShader { nullptr };  //!< 3d CPV solid-color shader
+        //!< Fallback shaders with CPV support
+        MHWRender::MShaderInstance* _fallbackCPVShaders[FallbackShaderTypeCount];
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2RenderDelegate/render_delegate.cpp:356]: Can you initialize the array elements to nullptr?

**Review by JGamache-autodesk — APPROVED** [review]

File: lib/mayaUsd/render/vp2RenderDelegate/render_delegate.cpp (hunk 1 above)
Comment (HdC-adsk) [reply lib/mayaUsd/render/vp2RenderDelegate/render_delegate.cpp]: Good point. Thanks.

**Review by williamkrick — APPROVED** [review]

**Comment by HdC-adsk** [comment]
@kxl-adsk The PR is ready for merge because the Maya Preview Release 120 is now live on the beta site.

**Pull request merged**